inc/search.class.php

```diff
@@ -2761,27 +2764,37 @@ static function addDefaultWhere($itemtype) {
 
                $condition .= ") ";
             }
-            return $condition;
+            break;;
 
          case 'Config':
             $availableContexts = array('core') + $_SESSION['glpi_plugins'];
             $availableContexts = implode("', '", $availableContexts);
             $condition = "`context` IN ('$availableContexts')";
-            return $condition;
+            break;
 
          default :
             // Plugin can override core definition for its type
             if ($plug = isPluginItemType($itemtype)) {
                $function = 'plugin_'.$plug['plugin'].'_addDefaultWhere';
                if (function_exists($function)) {
-                  $out = $function($itemtype);
-                  if (!empty($out)) {
-                     return $out;
-                  }
+                  $condition = $function($itemtype);
                }
             }
-            return "";
       }
+
+      /* Hook to restrict user right on current itemtype */
+      /* @since 9.2, see CommonDBTM::can for code sample */
+      $item = getItemForItemtype($itemtype);
+      Plugin::doHook("item_can", $item);
+      if (isset($item->add_where)) {
```

Comment:
Is the add_where property already used? If not, maybe it should be reset once handled, no?